internal/agent/device/applications/provider/image.go

```diff
@@ -137,6 +153,14 @@ func (p *imageProvider) Verify(ctx context.Context) error {
 		}
 	}()
 
+	// For artifacts, use stricter validation
+	if ociType == dependency.OCITypeArtifact && p.spec.AppType == v1alpha1.AppTypeQuadlet {
```

Comment:
I'm not sure I understand this check. Should the Verifys in the new handlers just always do this? My expectation at the end of the day would be that all formats: image, artifact, inline, and embedded would always setup a directory in the exact same way

i.e.
```
quadlet.container
quadlet.image
...
path/to/conf.conf
container.d/1-dropin.conf
```

What specifically about artifacts makes it need additional validation? I think the extra validations you added are more comprehensive and probably should just be applied to all types.